TextFieldEffects/TextFieldEffects/HoshiTextField.swift

```diff
@@ -12,178 +12,226 @@ import UIKit
+  private var inactivePlaceholderPoint: CGPoint {
+    get {
+      if let font = font, font.pointSize > 15 {
+        return CGPoint(x: 0, y: 14)
+      } else {
+        return CGPoint(x: 0, y: 20)
+      }
+    }
+  }
 
+  private var placeholderLabelOriginalText: String?
+  private var placeholderFont: UIFont?
+  private var height: CGFloat = 45
+
+  open func showError(message: String) {
+    placeholderLabelOriginalText = placeholderLabel.text
+    placeholderLabel.textColor = borderActiveColor
+    placeholderLabel.text = message
+    activeBorderLayer.frame = self.rectForBorder(self.borderThickness.active)
+    activeBorderLayer.isHidden = false
+    placeholderLabel.sizeToFit()
+  }
+
+  open func hideError() {
+    placeholderLabel.text = placeholderLabelOriginalText != nil ? placeholderLabelOriginalText : placeholderLabel.text
+    placeholderLabel.textColor = placeholderColor
+    activeBorderLayer.isHidden = true
+    placeholderLabel.sizeToFit()
+  }
+
+  // MARK: - TextFieldEffects
+
+  override open func drawViewsForRect(_ rect: CGRect) {
+    frame = CGRect(x: frame.origin.x,
+                   y: frame.origin.y,
+                   width: rect.size.width ,
+                   height: height)
+    configurePlaceholderLabelFrame()
+    configurePlaceholderFont()
+    updateBorder()
+    updatePlaceholder()
+
+    layer.addSublayer(inactiveBorderLayer)
+    layer.addSublayer(activeBorderLayer)
+    addSubview(placeholderLabel)
+  }
+
+  override open func animateViewsForTextEntry() {
+    if text!.isEmpty {
+      UIView.animate(withDuration: 0.35,
+                     delay: 0.0,
+                     usingSpringWithDamping: 0.8,
+                     initialSpringVelocity: 1.0,
+                     options: .beginFromCurrentState,
+                     animations: ({
+                      self.placeholderLabel.transform = CGAffineTransform(scaleX: 0.75, y: 0.75)
+                      self.placeholderLabel.font = self.placeholderFont
+                      self.placeholderLabel.sizeToFit()
+                      self.placeholderLabel.frame.origin = self.activePlaceholderPoint
+                     }), completion: { _ in
+                      self.animationCompletionHandler?(.textEntry)
+
+      })
     }
-    
-    // MARK: - Overrides
-    
-    override open func editingRect(forBounds bounds: CGRect) -> CGRect {
-        return bounds.offsetBy(dx: textFieldInsets.x, dy: textFieldInsets.y)
+  }
+
+  override open func animateViewsForTextDisplay() {
+    if text!.isEmpty {
+      UIView.animate(withDuration: 0.35, delay: 0.0,
+                     usingSpringWithDamping: 0.8,
+                     initialSpringVelocity: 2.0,
+                     options: UIViewAnimationOptions.beginFromCurrentState,
+                     animations: ({
+                      self.placeholderLabel.transform = CGAffineTransform(scaleX: 1.0, y: 1.0)
+                      guard let fontSize = self.font?.pointSize else { return }
```

Comment:
Same comment as before. But why are you creating new font and not just assigning it directly like `self.placeholderLabel.font = self.font`?